### PR TITLE
Dev

### LINKED ISSUE #23: Elasticsearch installation error
i'm encountering some error while setting up GreedyBear locally
After doing the `docker-compose -p greedybear up` cmd.
It originates from settings.py where Elasticsearch client is being initialized. The ELASTIC_ENDPOINT variable in my env file is empty
Screenshot from 2022-01-02 19-51-22

### PR DESCRIPTION
**Pull request by devmrfitz**
# Description

Added elasticsearch container for development

## Related issues
closes #23 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [X] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [X] The pull request is for the branch `dev`
- [X] The tests gave 0 errors.
- [X] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

### CODE REVIEW

**Comment by devmrfitz** [comment]
@mlodic Please review this PR

**Review by mlodic — CHANGES_REQUESTED** [review]
I won't put that default in the `ELASTIC_ENDPOINT` because that would also be inherited by new production deployments but that is only for development so it should not be a default. On the contrary, the right documentation should describe this case and how to handle it.

Then, ok the `elasticsearch` instance but there is something else to consider that I did not mention in the #23.
First, we would need `kibana` too, otherwise it would be really difficult to perform manual tests on ElasticSearch. Second, this would be anyway an optional configuration that should be in a new additional compose file: this is because if I am running a local instance I could still want to connect to a real T-Pot instance and, in that case, this container is not required

**Comment by devmrfitz** [comment]
@mlodic Please have a look now

File: docker/default.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ services:
       - env_file
     depends_on:
       - postgres
+      - elasticsearch
```
Comment (mlodic) [inline docker/default.yml]: this line should be added as a dependency only in the docker/elasticsearch.yml file. It is just a matter of adding uwsgi again with just this line

File: README.md [hunk 2]
```diff
@@ -83,4 +83,9 @@ docker-compose -p greedybear up
 
 # shut down the application
 docker-compose -p greedybear down
-```
\ No newline at end of file
+```
+Note that GreedyBear *needs* a running instance of ElasticSearch to function.
+If you don't have one, you can make the following changes to make GreeyBear spin up it's own ElasticSearch and Kibana instances:
+
+1) In ```docker/env_file```, set the variable ```ELASTICSEARCH_HOST``` to ```http://elasticsearch:9200```.
```
Comment (mlodic) [inline README.md]: the name is different in that file `ELASTIC_ENDPOINT`

File: README.md [hunk 3]
```diff
@@ -83,4 +83,9 @@ docker-compose -p greedybear up
 
 # shut down the application
 docker-compose -p greedybear down
-```
\ No newline at end of file
+```
+Note that GreedyBear *needs* a running instance of ElasticSearch to function.
```
Comment (mlodic) [inline README.md]: "Note that GreedyBear *needs* a running instance of ElasticSearch to function. This should be the ElasticSearch of a TPoT."

File: README.md [hunk 4]
```diff
@@ -83,4 +83,9 @@ docker-compose -p greedybear up
...
+If you don't have one, you can make the following changes to make GreeyBear spin up it's own ElasticSearch and Kibana instances:
+
```
Comment (mlodic) [inline README.md:90]: "...Care! This option would require enough RAM to run the additional containers. Suggested is >=16GB."

**Review by mlodic — COMMENTED** [review]
👍🏻

**Comment by devmrfitz** [comment]
@mlodic Fixed all the mentioned issues

File: docker/elasticsearch.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+version: '3.8'
+
+services:
+  uwsgi:
```
Comment (mlodic) [inline docker/elasticsearch.yml:4]: it works as a "override" so no need to replicate what already is in the `default.yml`

just uwsgi, depend_on and elasticsearch are enough

Comment (devmrfitz) [reply docker/elasticsearch.yml:4]: Do I need to mention ```postgres``` in ```depends_on``` again too?

**Comment by devmrfitz** [comment]
@mlodic resolved

**Review by mlodic — APPROVED** [review]

**Pull request merged**